ffi/diplomat/src/timezone.rs

```diff
@@ -31,11 +31,8 @@ pub mod ffi {
         #[diplomat::rust_link(icu::timezone::CustomTimeZone::from_str, FnInStruct)]
         #[diplomat::rust_link(icu::timezone::GmtOffset::from_str, FnInStruct, hidden)]
         pub fn create_from_string(s: &str) -> DiplomatResult<Box<ICU4XCustomTimeZone>, ICU4XError> {
-            // TODO(#2543): Use a byte parsing API once available in CustomTimeZone (also #2520)
-            if str::from_utf8(s.as_bytes()).is_err() {
-                return Err(ICU4XError::TimeZoneInvalidOffsetError).into();
-            }
-            CustomTimeZone::from_str(s)
+            let bytes = s.as_bytes(); // #2520
```

Comment:
```suggestion
            let bytes = s.as_bytes();
```

we don't need that issue breadcrumb anymore since we've moved to a byte parsing API